From bccdcc4e491e33e1e006f4c1a443913065f07f24 Mon Sep 17 00:00:00 2001
From: Dimitar Dimitrov <dimitar@dinux.eu>
Date: Sun, 15 Apr 2018 15:03:27 +0300
Subject: [PATCH 05/13] testsuite: Add check for unsupported TI ABI PRU
 features to testsuite

Not all C language features are supported when -mabi=ti option is
used for PRU target.

gcc/testsuite/ChangeLog:

2018-04-15  Dimitar Dimitrov  <dimitar@dinux.eu>

	* lib/gcc-dg.exp: Filter unsupported features in PRU's TI ABI mode.
	* lib/target-utils.exp: Ditto.

Signed-off-by: Dimitar Dimitrov <dimitar@dinux.eu>
---
 gcc/testsuite/lib/gcc-dg.exp       | 11 +++++++++++
 gcc/testsuite/lib/target-utils.exp |  8 ++++++++
 2 files changed, 19 insertions(+)

diff --git a/gcc/testsuite/lib/gcc-dg.exp b/gcc/testsuite/lib/gcc-dg.exp
index c5ab850c840..f4976520282 100644
--- a/gcc/testsuite/lib/gcc-dg.exp
+++ b/gcc/testsuite/lib/gcc-dg.exp
@@ -388,6 +388,17 @@ proc gcc-dg-prune { system text } {
 	return "::unsupported::memory full"
     }
 
+    if {[string match "pru-*" $system] && \
+	    [string match "*error: function pointers not supported with -mabi=ti option*" $text]} {
+	# The format here is important.  See dg.exp.
+	return "::unsupported::abi"
+    }
+    if {[string match "pru-*" $system] && \
+	    [string match "*error: large return values not supported with -mabi=ti option*" $text]} {
+	# The format here is important.  See dg.exp.
+	return "::unsupported::abi"
+    }
+
     return $text
 }
 
diff --git a/gcc/testsuite/lib/target-utils.exp b/gcc/testsuite/lib/target-utils.exp
index 732a1827a02..e288447ef1d 100644
--- a/gcc/testsuite/lib/target-utils.exp
+++ b/gcc/testsuite/lib/target-utils.exp
@@ -44,5 +44,13 @@ proc ${tool}_check_unsupported_p { output } {
 	     [string match "*exceeds local store*" $output] } {
 	return "memory full"
     }
+    if {[istarget pru-*] && \
+	    [string match "*error: large return values not supported with -mabi=ti option*" $output]} {
+	return "abi not supported"
+    }
+    if {[istarget pru-*] && \
+	    [string match "*error: function pointers not supported with -mabi=ti option*" $output]} {
+	return "abi not supported"
+    }
     return ""
 }
-- 
2.11.0

